### PR TITLE
Rechargers and IPCs Quality of Life

### PR DESCRIPTION
**Pull request by VisVirific**
* You can now shove in IPCs via grabs or drag+drop into the rechargers
* While it doesn't take 5 billions years like stationbounds, it still takes around 5 seconds
* I am very sorry if the code looks weird. I am very sleep deprived

### CODE REVIEW

**Comment by VisVirific** [comment]
!review

File: code/game/machinery/rechargestation.dm [hunk 1]
```diff
@@ -295,4 +307,30 @@
 		else
 			to_chat(user, SPAN_DANGER("Cancelled loading [C] into the charger. You and [C] must stay still!"))
 		return
+	
+	else if(istype(C, /mob/living/carbon/human/machine)) // IPCs don't take as long
```
Comment (mikomyazaki) [inline code/game/machinery/rechargestation.dm:311]:
```suggestion
	else if(isipc(C)) // IPCs don't take as long
```

This helper would probably be better here.

File: code/game/machinery/rechargestation.dm [hunk 2]
```diff
@@ -154,6 +154,18 @@
 		else if(default_part_replacement(user, O))
 			return
 
+	if(istype(O, /obj/item/grab))
+		var/obj/item/grab/grab = O
+		var/mob/living/L = grab.affecting
+		if(!L.isSynthetic())
+			return
+		
+		user.visible_message(SPAN_NOTICE("[user] starts putting [L] into [src]."), SPAN_NOTICE("You start putting [L] into [src]."), range = 3)
```
Comment (mikomyazaki) [inline code/game/machinery/rechargestation.dm:163]: The code below here that you added does the same thing as some of the code you added to `/obj/machinery/recharge_station/MouseDrop_T` -- But you have better error messages in `MouseDrop_T`. You should extract all the common code into another proc and call it in both locations.

**Review by mikomyazaki — CHANGES_REQUESTED** [review]

File: code/game/machinery/rechargestation.dm (hunk 1 above)
Comment (VisVirific) [reply code/game/machinery/rechargestation.dm]: Done!

File: code/game/machinery/rechargestation.dm (hunk 2 above)
Comment (VisVirific) [reply code/game/machinery/rechargestation.dm]: Done as well. It should be better now

**Review by mikomyazaki — APPROVED** [review]

**Pull request merged**